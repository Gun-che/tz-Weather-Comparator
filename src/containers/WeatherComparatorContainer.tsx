import React, { Dispatch } from 'react'
import { connect, ConnectedProps } from 'react-redux'
import { IStoreState } from '../types/state'
import { createWCRequest } from '../actions/wc'
import { IAction, IWeatherComparatorRequestArgs } from '../types/actions'
import { WeatherComparator } from '../pages/WeatherComparator'


const mapStateToProps = (state: IStoreState) => ({
  data: state.wc.data,
  isFetching: state.wc.isFetching,
  message: state.wc.message,
  code: state.wc.code,
})

const mapDispatchToProps = (dispatch: Dispatch<IAction>) => ({
  getWeather: (data: IWeatherComparatorRequestArgs) => dispatch(createWCRequest(data)),
})

const connector = connect(mapStateToProps, mapDispatchToProps)

export type PropsFromRedux = ConnectedProps<typeof connector>

export const WeatherComparatorContainer: React.FC<PropsFromRedux> = ({
  getWeather,
  data,
  isFetching,
  message,
  code,
}) => {

  return (
    <div>
      <WeatherComparator
        getWeather={getWeather}
        data={data}
        isFetching={isFetching}
        message={message}
        code={code}
      />
    </div>
  )
}

export default connector(WeatherComparatorContainer)
