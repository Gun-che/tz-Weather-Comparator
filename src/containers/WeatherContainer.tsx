import React, { Dispatch } from 'react'
import { connect, ConnectedProps } from 'react-redux'
import { IStoreState } from '../types/state'
import { createWeatherRequest } from '../actions/weather'
import { IAction, IWeatherRequestArgs } from '../types/actions'
import { Weather } from '../pages/Weather'


const mapStateToProps = (state: IStoreState) => ({
  data: state.weather.data,
  isFetching: state.weather.isFetching,
  message: state.weather.message,
  code: state.weather.code,
  daily: state.weather.dailyForecastData,
  hourly: state.weather.hourlyForecastData,
})

const mapDispatchToProps = (dispatch: Dispatch<IAction>) => ({
  getWeather: (data: IWeatherRequestArgs) => dispatch(createWeatherRequest(data)),
})

const connector = connect(mapStateToProps, mapDispatchToProps)

export type PropsFromRedux = ConnectedProps<typeof connector>
// type Props = PropsFromRedux & {background?: string}
// or
// interface IProps extends ConnectedProps<typeof connector> {
//   background?: string;
// } 

export const WeatherContainer: React.FC<ConnectedProps<typeof connector>> = ({
  getWeather,
  data,
  isFetching,
  message,
  code,
  daily,
  hourly
}) => {
  return (
    <div>
      <Weather
        daily={daily}
        hourly={hourly}
        getWeather={getWeather}
        data={data}
        isFetching={isFetching}
        message={message}
        code={code}
      />
    </div>
  )
}


export default connector(WeatherContainer)
