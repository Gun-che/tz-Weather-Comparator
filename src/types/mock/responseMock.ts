import { IDailyWeatherData, IHourlyWeatherData, IWeatherData, IAllWeatherData } from "../apiResponse";

export const weatherDataMock: IWeatherData = {
  name: '1',
  visibility: 1,
  weather: [{
    id: 1,
    icon: '1',
    description: '1',
  }],
  wind: { speed: 1 },
  coord: { lon: 1, lat: 1 },
  main: {
    feels_like: 1,
    humidity: 1,
    pressure: 1,
    temp: 1,
  },
}

export const hourlyWeatherDataMock: IHourlyWeatherData = {
  clouds: 1,
  feels_like: 1,
  humidity: 1,
  pressure: 1,
  temp: 1,
  weather: [{
    id: 1,
    icon: '1',
    description: '1',
  }],
  wind_speed: 1,
  dt: 1,
}

export const dailyWeatherDataMock: IDailyWeatherData = {
  clouds: 10,
  weather: [{
    id: 1,
    icon: '1',
    description: '1',
  }],
  pressure: 10,
  feels_like: { day: 1, night: 1, eve: 1, morn: 1 },
  humidity: 1,
  sunrise: 1,
  sunset: 1,
  temp: { day: 1, night: 1, eve: 1, morn: 1 },
  wind_speed: 1,
  dt: 1,
}

export const allWeatherDataMock: IAllWeatherData = {
  hourly: [hourlyWeatherDataMock],
  daily: [dailyWeatherDataMock],
}