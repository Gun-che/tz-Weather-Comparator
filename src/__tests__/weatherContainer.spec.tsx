import React from 'react'
import { shallow } from 'enzyme'
import { WeatherContainer } from '../containers/WeatherContainer'
import { weatherDataMock, hourlyWeatherDataMock, dailyWeatherDataMock } from '../types/mock/responseMock'

describe('Weather container', () => {
  const props = {
    getWeather: jest.fn(),
    data: [],
    isFetching: false,
    message: '1',
    code: 1,
    daily: [],
    hourly: [],
  }

  describe('Weather container init', () => {
    const WeatherContainerRender = shallow(<WeatherContainer {...props} />)

    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })

  describe('Weather container render <Weather>', () => {
    const nextProps = {
      ...props,
      data: [weatherDataMock]
    }

    const WeatherContainerRender = shallow(<WeatherContainer {...nextProps} />)

    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })


  describe('Weather container render <Weather>', () => {
    const nextProps = {
      ...props,
      hourly: [hourlyWeatherDataMock, hourlyWeatherDataMock]
    }

    const WeatherContainerRender = shallow(<WeatherContainer {...nextProps} />)

    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })


  describe('Weather container render <Weather>', () => {
    const nextProps = {
      ...props,
      daily: [dailyWeatherDataMock, dailyWeatherDataMock]
    }

    const WeatherContainerRender = shallow(<WeatherContainer {...nextProps} />)

    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })


  describe('Weather container with error', () => {
    const nextProps = {
      ...props,
      msg: 'Something going wrong',
    }

    const WeatherContainerRender = shallow(<WeatherContainer {...nextProps} />)

    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })


  describe('Weather container loading', () => {
    const nextProps = {
      ...props,
      isFetching: true,
    }

    const WeatherContainerRender = shallow(<WeatherContainer {...nextProps} />)
    it('renders properly', () => {
      expect(WeatherContainerRender).toMatchSnapshot()
    })
  })
})


