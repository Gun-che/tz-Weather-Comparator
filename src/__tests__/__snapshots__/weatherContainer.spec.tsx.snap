// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather container Weather container init renders properly 1`] = `
<div>
  <Weather
    code={1}
    daily={Array []}
    data={Array []}
    getWeather={[MockFunction]}
    hourly={Array []}
    isFetching={false}
    message="1"
  />
</div>
`;

exports[`Weather container Weather container loading renders properly 1`] = `
<div>
  <Weather
    code={1}
    daily={Array []}
    data={Array []}
    getWeather={[MockFunction]}
    hourly={Array []}
    isFetching={true}
    message="1"
  />
</div>
`;

exports[`Weather container Weather container render <Weather> renders properly 1`] = `
<div>
  <Weather
    code={1}
    daily={Array []}
    data={
      Array [
        Object {
          "coord": Object {
            "lat": 1,
            "lon": 1,
          },
          "main": Object {
            "feels_like": 1,
            "humidity": 1,
            "pressure": 1,
            "temp": 1,
          },
          "name": "1",
          "visibility": 1,
          "weather": Array [
            Object {
              "description": "1",
              "icon": "1",
              "id": 800,
            },
          ],
          "wind": Object {
            "speed": 1,
          },
        },
      ]
    }
    getWeather={[MockFunction]}
    hourly={Array []}
    isFetching={false}
    message="1"
  />
</div>
`;

exports[`Weather container Weather container render <Weather> renders properly 2`] = `
<div>
  <Weather
    code={1}
    daily={Array []}
    data={Array []}
    getWeather={[MockFunction]}
    hourly={
      Array [
        Object {
          "clouds": 1,
          "dt": 1,
          "feels_like": 1,
          "humidity": 1,
          "pressure": 1,
          "temp": 1,
          "weather": Array [
            Object {
              "description": "1",
              "icon": "1",
              "id": 800,
            },
          ],
          "wind_speed": 1,
        },
        Object {
          "clouds": 1,
          "dt": 1,
          "feels_like": 1,
          "humidity": 1,
          "pressure": 1,
          "temp": 1,
          "weather": Array [
            Object {
              "description": "1",
              "icon": "1",
              "id": 800,
            },
          ],
          "wind_speed": 1,
        },
      ]
    }
    isFetching={false}
    message="1"
  />
</div>
`;

exports[`Weather container Weather container render <Weather> renders properly 3`] = `
<div>
  <Weather
    code={1}
    daily={
      Array [
        Object {
          "clouds": 10,
          "dt": 1,
          "feels_like": Object {
            "day": 1,
            "eve": 1,
            "morn": 1,
            "night": 1,
          },
          "humidity": 1,
          "pressure": 10,
          "sunrise": 1,
          "sunset": 1,
          "temp": Object {
            "day": 1,
            "eve": 1,
            "morn": 1,
            "night": 1,
          },
          "weather": Array [
            Object {
              "description": "1",
              "icon": "1",
              "id": 800,
            },
          ],
          "wind_speed": 1,
        },
        Object {
          "clouds": 10,
          "dt": 1,
          "feels_like": Object {
            "day": 1,
            "eve": 1,
            "morn": 1,
            "night": 1,
          },
          "humidity": 1,
          "pressure": 10,
          "sunrise": 1,
          "sunset": 1,
          "temp": Object {
            "day": 1,
            "eve": 1,
            "morn": 1,
            "night": 1,
          },
          "weather": Array [
            Object {
              "description": "1",
              "icon": "1",
              "id": 800,
            },
          ],
          "wind_speed": 1,
        },
      ]
    }
    data={Array []}
    getWeather={[MockFunction]}
    hourly={Array []}
    isFetching={false}
    message="1"
  />
</div>
`;

exports[`Weather container Weather container with error renders properly 1`] = `
<div>
  <Weather
    code={1}
    daily={Array []}
    data={Array []}
    getWeather={[MockFunction]}
    hourly={Array []}
    isFetching={false}
    message="1"
  />
</div>
`;
